vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/SampleOverviewComponent.java

```diff
@@ -278,26 +293,32 @@ private boolean hasExperimentalGroupsDefined() {
 
     private void registerSamplesListener() {
       registerBatchButton.addClickListener(event -> {
-        if(hasExperimentalGroupsDefined()) {
-          sampleRegistrationDialog.open();
+        if (hasExperimentalGroupsDefined()) {
+          batchRegistrationDialog.open();
         } else {
-          InformationMessage infoMessage = new InformationMessage("No experimental groups are defined",
+          InformationMessage infoMessage = new InformationMessage(
+              "No experimental groups are defined",
               "You need to define experimental groups before adding samples.");
           StyledNotification notification = new StyledNotification(infoMessage);
           notification.open();
         }
       });
-      //ToDo Replace with received samples from SampleInformationService
       showEmptyViewButton.addClickListener(event -> showEmptyView());
     }
 
-    private void configureBatchRegistrationDialog() {
-      sampleRegistrationDialog.addSampleRegistrationEventListener(event -> {
-        sampleRegistrationDialog.resetAndClose();
-        showSamplesView();
-      });
-      sampleRegistrationDialog.addCancelEventListener(
-          event -> sampleRegistrationDialog.resetAndClose());
+    private void addEventListeners() {
+      batchRegistrationDialog.addBatchRegistrationEventListener(
+          batchRegistrationEvent -> registerBatch(
+              batchRegistrationEvent.getSource().batchRegistrationContent()).onValue(batchId -> {
+            List<SampleRegistrationContent> sampleRegistrationContents = batchRegistrationEvent.getSource()
+                .sampleRegistrationContent();
+            registerSamplesForBatch(batchId, sampleRegistrationContents).onValue(result -> {
+              batchRegistrationDialog.resetAndClose();
+              displayRegistrationSuccess();
+            }).onError(responseCode -> displayRegistrationFailure());
+          }).onError(responseCode -> displayRegistrationFailure()));
+      batchRegistrationDialog.addCancelEventListener(
+          event -> batchRegistrationDialog.resetAndClose());
```

Comment:
This is very difficult to read and understand.  I suggest to extract partials to own functions.

Reply:
Thanks for the feedback i slimmed it down :)